compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -83,6 +83,9 @@ pub enum MirPhase {
     ///    access to. This occurs in generator bodies. Such locals do not behave like other locals,
     ///    because they eg may be aliased in surprising ways. Runtime MIR has no such special locals -
     ///    all generator bodies are lowered and so all places that look like locals really are locals.
+    ///  - Intrinsics: In analysis MIR, intrinsics are typically represented as [`TerminatorKind::Call`]s
+    ///    to their `extern "rust-intrinsic"` declarations. In runtime MIR, some intrinsics are lowered
+    ///    to MIR constructs not used earlier, such as `mem::transmute` → [`CastKind::Transmute`].
```

Comment:
Some of these MIR constructs are used earlier though, so this statement in its generality seems misleading?

Reply:
I'm trying to reflect the phase difference first brought up in <https://github.com/rust-lang/rust/pull/108442#discussion_r1118188243>.

I'll happily change this to any text that makes you and @JakobDegen content; let me know what you'd like to see.

Reply:
From that perspective this is not about intrinsics at all then, but about the specific CastKind::Transmute.

Does this list exhaustively list all cases where an operation is only allowed/forbidden in some phases? The other items in the list are things where the same syntax means different things in different phases; that is different from CastKind::Transmute which always means the same thing when it appears, but is not always allowed to appear.

Reply:
Ah, I like that argument!  I'll drop the new section from here, in favour of just the note on `CastKind::Transmute` itself.